File: order-mgmt/src/main/kotlin/com/order/ordermgmt/models/Manager.kt
package com.order.ordermgmt.models

import javax.persistence.Entity
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne


@Entity
class Manager(
        email:String,
        password:String,
        username: String,
        phoneNumber: String,
        @ManyToOne
        @JoinColumn(name = "restaurant_id")
        val restaurant:Restaurant
        ): User(email,password,username,phoneNumber) {

}

File: restaurant_app/lib/models/manager.dart

import 'package:common_widgets/models/user.dart';
import 'package:flutter/widgets.dart';
import 'package:meta/meta.dart';

class Manager extends User {
  int restaurant;

  Manager(
      {@required id,
      @required email,
      @required username,
      @required phoneNumber,
      @required this.restaurant})
      : super(id: id, email: email, username: username, phoneNumber: phoneNumber);
            
  static Manager fromMap(Map<String, dynamic> map) {
    return Manager(
        id: map['id'],
        email: map['email'],
        username: map['username'],
        phoneNumber: map['phoneNumber'],
        restaurant: map['restaurant']['id']);
  }
}
